### DISCUSSION TITLE
WebUI

### DISCUSSION THREAD

**Discussion by snickers2k** [meisnate12/Plex-Meta-Manager#1737] [General]
Hey. Is there any chance PMM gets a WebUI anytime soon? 
I don't update my PMM configs, already for 2 years, after i had somewhat of an "okayish" setup. 
Because it was just too overwhelming with all the options and stuff .. even the Wiki is complete overkill :D 

i mean, don't get me wrong, it's awesome PMM came this far and is this big of an improvement for Plex, but i really wish there would be an WebUI for easier maintenance. 

i bet this would skyrocket the amount of PMM users

So before going back into config editing , i thought, i ask for the status :)

**Comment by chazlarson** [comment]
A GUI is a nice-to-have, but it is absolutely not trivial to implement, and is probably a long way away.

It becomes another thing to keep up to date, it becomes another thing that users need to configure and run, it's just not the silver bullet that most think it is.

I don't understand what you mean by "even the Wiki is complete overkill".  PMM is a complicated tool, and it's "complete overkill" to document its capabilities?

Should we keep half its capabilities secret?  Remove all the examples?  What would be a more reasonable level of documentation if there is currently too much of it?

**Comment by benniblot** [comment]
Have there been any discussions about the tech stack? I could see myself creating a simple config editor which dynamically explains the options, which could later be implemented into a potential fully-featured web ui. Just a config editor/creator alone would make Kometa in my opinion a lot more accessible. It could work of the yaml schema file, which would mean little to none maintenance needed, aside from updating the dependencies from time to time.

**Reply by YozoraXCII** [reply]
There is a tool in development which aims to make the creation of config.yml files easier with a visual UI. It has somewhat stalled recently due to technical difficulties but hopefully it can be picked back up as it looked really promising from the work done so far.

**Reply by benniblot** [reply]
Is it going to be open sourced? I’d love to contribute

**Reply by YozoraXCII** [reply]
Yes I believe that is the intention if we can get it off the ground. it isn't being developed by the owner of Kometa (Sohjiro/meisnate12) but instead by a couple people in our team.

**Reply by benniblot** [reply]
Great, can’t wait to hear more about it

**Comment by vednolacni** [comment]
any update on this?

**Reply by YozoraXCII** [reply]
In terms of an actual Kometa GUI, no. We have a working tool to help users create their config.yml files, it is in beta testing with our sponsors at the moment and we're making really good progress on it.
